### PR TITLE
refactor(core): remove deprecated APIs for v11

### PR DESCRIPTION
**Pull request by crisbeto**
Removes the deprecated APIs for v11 from `material/core`. Also fixes that the MDC-based form field doesn't support the global `floatLabel` option.

BREAKING CHANGES:
* `MAT_LABEL_GLOBAL_OPTIONS` from `material/core` has been removed. Import `MAT_FORM_FIELD_DEFAULT_OPTIONS` from `material/form-field` instead.
* `FloatLabelType` from `material/core` has been removed. Import `FloatLabelType` from `material/form-field` instead.
* `LabelOptions` from `material/core` has been removed. Import `MatFormFieldDefaultOptions` from `material/form-field` instead.
* `MAT_DATE_LOCALE_PROVIDER` has been removed, because it is no longer needed since MAT_DATE_LOCALE has been changed to a scoped injectable. If you are importing and providing this in your code you can simply remove it.
* `MatLineSetter` class has been removed. Use the `setLines` function instead.
* `hammer` property from `GranularSanityChecks` has been removed, because it isn't being used anymore.
* `document` parameter of the `MatCommonModule` constructor is now required.

### CODE REVIEW

File: src/material/form-field/form-field.ts [hunk 1]
```diff
@@ -284,14 +281,19 @@ export class MatFormField extends _MatFormFieldMixinBase
 
   constructor(
       public _elementRef: ElementRef, private _changeDetectorRef: ChangeDetectorRef,
-      @Optional() @Inject(MAT_LABEL_GLOBAL_OPTIONS) labelOptions: LabelOptions,
+      /**
+       * @deprected `_labelOptions` parameter no longer being used. To be removed.
```
Comment (devversion) [inline src/material/form-field/form-field.ts:285]: It's a bit unfortunate that we didn't deprecate this parameter (my bad obviously!)

```suggestion
       * @deprecated `_labelOptions` parameter no longer being used. To be removed.
```

**Review by devversion — APPROVED** [review]
LGTM. one minor comment

**Review by jelbourn — APPROVED** [review]
LGTM

**Comment by wagnermaciel** [comment]
@crisbeto Please resolve this merge conflict

**Comment by crisbeto** [comment]
Rebased.

**Comment by annieyw** [comment]
@crisbeto I think the recently merged mdc_select was using `MAT_LABEL_GLOBAL_OPTIONS` can you switch it to `MAT_FORM_FIELD_DEFAULT_OPTIONS`?

**Comment by crisbeto** [comment]
Fixed.

**Pull request merged**